src/fromager/requirements_file.py

```diff
@@ -0,0 +1,22 @@
+import collections
+import logging
+import pathlib
+import typing
+
+logger = logging.getLogger(__name__)
+
+
+def parse_requirements_file(
+    req_files: typing.Iterable[pathlib.Path],
+) -> dict[str, typing.Iterable[str]]:
```

Comment:
Why does this return a dict?

Reply:
It feels like the fact that there might be multiple files and the requirements from each may need to be kept distinct is a problem for the caller to deal with. This function would be simpler if it only dealt with 1 file at a time. How many places do we have where it's called with multiple files, though? Is that the more common pattern?

Reply:
This function is used only twice: once in bootstrap.py to parse all the requirement files and once in constraint.py to parse a constraint file. In bootstrap.py it is called with multiple files.

Can't really tell from this if it will be a common pattern since it is used only twice.

Reply:
OK, let's err on the side of keeping the parsing function simple and move the complexity to the caller. If we end up needing to deal with multiple files in more places, we can add another function that wraps this new one.